content/3_Silver/Func_Graphs.mdx

```diff
@@ -716,6 +716,95 @@ print(" ".join(str(i + 1) for i in ans))
 </PySection>
 </LanguageSection>
 
+### Solution 3
+
+It's also possible to use `floyd(x)` to generate answers for all vertices in the connected component
+containing `x` without using adjacency lists.
+
+<LanguageSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Badge {
+	static int[] ans;
+	static int[] arr;
+	public static void main(String[] args) throws IOException {
+		BufferedReader x = new BufferedReader(new InputStreamReader(System.in));
+		StringTokenizer st = new StringTokenizer(x.readLine());
+		int n = Integer.parseInt(st.nextToken());
+		arr = new int[n + 1];
+		st = new StringTokenizer(x.readLine());
+		for (int i = 1; i <= n; i++) {
+			arr[i] = Integer.parseInt(st.nextToken());
+		}
+
+		ans = new int[n + 1];
+		Arrays.fill(ans, -1);
+		for (int i = 1; i <= n; i++) {
+			if (ans[i] == -1) { floyd(i); }
+		}
+
+		StringBuilder output = new StringBuilder();
+		for (int i = 1; i <= n; i++) { output.append(ans[i]).append(" "); }
+		output.setLength(output.length() - 1);
+		System.out.println(output);
+	}
+
+	/**
+	 * fills up all vertices from curr to firstCycle,
+	 * whose answer has already been calculated
+	 */
+	static void fill_up(int curr, int first_cycle) {
```

Comment:
use camelCase

Reply:
previously used camelCase, @ryanchou-dev  suggested to use snake case [above](https://github.com/cpinitiative/usaco-guide/pull/4336#discussion_r1518676540)

Reply:
oof uh very sorry about that, could you change it back

Reply:
updated